article.md

```diff
@@ -0,0 +1,265 @@
+    let (temp) = word_reverse_endian(value);
+    let (high, low) = unsigned_div_rem(temp, DIVISOR);
+    return (high, low);
+}
+// Changes input format to array of 64bit values in little endian format
+func populate_array_little_end{range_check_ptr, bitwise_ptr : BitwiseBuiltin*}(w0: felt, w1: felt, w2: felt, w3: felt, overflow: felt, signable_bytes : felt*){
+    
+    let (high, low) = split_to_64bit_small_endian_words(w0);
+    assert signable_bytes[0] = low;
+    assert signable_bytes[1] = high;
+
+    let (high, low) = split_to_64bit_small_endian_words(w1);
+    assert signable_bytes[2] = low;
+    assert signable_bytes[3] = high;
+
+    let (high, low) = split_to_64bit_small_endian_words(w2);
+    assert signable_bytes[4] = low;
+    assert signable_bytes[5] = high;
+
+    let (high, low) = split_to_64bit_small_endian_words(w3);
+    assert signable_bytes[6] = low;
+    assert signable_bytes[7] = high;
+
+    // Overflow is only 2 bytes long
+    let (temp) = word_reverse_endian(overflow);
+    let (value8, trash) = unsigned_div_rem(temp, 2**112);
+    assert signable_bytes[8] = value8;
+
+    return ();
+}
+```
+Now we can finish our `get_hash` function.
+```cario
+func get_hash{range_check_ptr,  keccak_ptr: felt*, bitwise_ptr : BitwiseBuiltin*}(starknet_address: felt, domain_hash : Uint256) -> (
+        hashed_msg : Uint256){
+
+    let (starknet_address_h, starknet_address_l) = split_felt(starknet_address);
+    
+    let (encoded_data : Uint256*) = alloc();
+    assert encoded_data[0] = Uint256(TYPE_HASH_LOW, TYPE_HASH_HIGH);
+    assert encoded_data[1] = Uint256(starknet_address_l, starknet_address_h);
+    let (data_hash) = keccak_uint256s_bigend{keccak_ptr=keccak_ptr}(2, encoded_data);
+
+    let prefix = PREFIX;
+    let (w0, prefix) = add_prefix(domain_hash.high, prefix);
+    let (w1, prefix) = add_prefix(domain_hash.low, prefix);
+    let (w2, prefix) = add_prefix(data_hash.high, prefix);
+    let (w3, overflow) = add_prefix(data_hash.low, prefix);
+    
+    let (signable_bytes : felt*) = alloc();
+    
+    // w0, w1, w2, w3, and overflow need to be split into 64 bit chunks in little endian format
+    populate_array_little_end(w0, w1, w2, w3, overflow, signable_bytes);
+
+    // final value is 66 bytes long 
+    let (res) = keccak_bigend{keccak_ptr=keccak_ptr}(signable_bytes, 66);
+    
+    return (hashed_msg = res);  
+}
+```
+All that's left to do is call `verify_eth_signature_uint256` in our `assert_valid_eth_signature` function.
+```cairo
+    verify_eth_signature_uint256{keccak_ptr=keccak_ptr}(hash_uint, r, s, v, eth_address);
+
+    finalize_keccak(keccak_ptr_start=keccak_ptr_start, keccak_ptr_end=keccak_ptr);
+
+    return ();
+}
+```
+We need to call `finaliza_keccak` because of the internal batching done.
+inside it. Failing to call `finalize_keccak()` will make the keccak function unsound - the prover will be able to choose any value as the keccak's result.
```

Comment:
There is something missing, new line starts with "inside it" and the previous line ends with a dot